osu.Game.Rulesets.Catch/Skinning/Argon/ArgonFruitPiece.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Utils;
+using osu.Game.Rulesets.Catch.Skinning.Default;
+using osu.Game.Rulesets.Catch.UI;
+using osuTK;
+
+namespace osu.Game.Rulesets.Catch.Skinning.Argon
+{
+    internal class ArgonFruitPiece : CatchHitObjectPiece
+    {
+        protected override Drawable HyperBorderPiece => hyperBorderPiece;
+
+        private Drawable hyperBorderPiece = null!;
+
+        private Container layers = null!;
+
+        private float rotationRandomness;
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            RelativeSizeAxes = Axes.Both;
+
+            int largeBlobSeed = RNG.Next();
```

Comment:
The usage of `RNG` rather than something like `StatelessRNG` means that if you play back a replay, let it play some, then rewind it and replay the rewound portion, the fruits will be different than they previously were if they left the screen. Unsure if bug or feature, just pointing it out as I see it.

Reply:
Right, this was one of my initial concerns. But on the flip-side, if it was to be stateless it will cause invalidation for every new usage of a pooled fruit. I'm not sure how much it matters to ensure consistency here.